### PR TITLE
Remove the need for go-bindata

### LINKED ISSUE #10075: Remove go-bindata
Use of `go-bindata` dates back from times when people ran Cilium as static binary. This has become uncommon and users either use the container image or a package manager which will both ship `/var/lib/cilium` directly so there is no need to unpack any assets via the binary.

### PR DESCRIPTION
**Pull request by tklauser**
Use of go-bindata dates back from times when people ran Cilium as static
binary. This has become uncommon and users either use the container
image or a package manager which will both ship /var/lib/cilium directly
so there is no need to unpack any assets via the binary.
    
For people still wanting to use Cilium as a static binary, e.g. for
local development provide the `install-bpf` Makefile target to install
the BPF assets into `/var/lib/cilium`.
    
This saves ~380 kB in the resulting cilium-agent binary:
    
      == daemon/cilium-agent ==
      bss                                7752192     7752160         -32
      data                                894041      651280     -242761
      dec                               64545230    64166071     -379159
      hex                                3d8e1ce     3d318b7      -5c917
      text                              55898997    55762631     -136366
    
Updates #10056
Fixes #10075

  ```release-note
go-bindata is no longer used to install BPF assets.
  ```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10177)

### CODE REVIEW

**Comment by tklauser** [comment]
test-me-please

File: daemon/daemon_main.go [hunk 1]
```diff
@@ -453,6 +453,7 @@ func init() {
 
 	flags.Bool(option.KeepBPFTemplates, false, "Do not restore BPF template files from binary")
 	option.BindEnv(option.KeepBPFTemplates)
+	flags.MarkDeprecated(option.KeepBPFTemplates, "This option is no longer supported and will be removed in v1.9")
```
Comment (tgraf) [inline daemon/daemon_main.go:461]: Awesome, please also add a paragraph to the upgrade guide. We maintain a section with all options that are being deprecated and what action users have to take.

**Review by tgraf — CHANGES_REQUESTED** [review]

File: daemon/daemon_main.go (hunk 1 above)
Comment (tklauser) [reply daemon/daemon_main.go:461]: Updated commit 035bc332e9dfff907f5347e3367a3b394e910099 ("option: mark --keep-bpf-templates as deprecated") to include a paragraph in the upgrade guide.

**Comment by tklauser** [comment]
test-me-please
test-docs-please

**Comment by tklauser** [comment]
test-docs-please

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-me-please

**Comment by brb** [comment]
CI failed due to well known flakes: `K8sServicesTest_Checks_ClusterIP_Connectivity_IPv6_Connectivity_Checks_service_on_same_node` and `K8sPolicyTest_Basic_Test_Validate_to-entities_policies_Validate_toEntities_All`. Re-running.

**Comment by brb** [comment]
test-me-please

**Review by brb — COMMENTED** [review]
Nice, a productivity booster!

Did you check what error will returned to a user when running cilium-agent as a standalone binary and w/o the bpf files installed?

Also, we should probably add a GSG for running cilium as standalone (not in this PR).

**Comment by tklauser** [comment]
> Did you check what error will returned to a user when running cilium-agent as a standalone binary and w/o the bpf files installed?

```
level=fatal msg="BPF Verifier: NOT OK. Unable to run checker for bpf_features" error="fork/exec /var/lib/cilium/bpf/run_probes.sh: no such file or directory" subsys=linux-datapath
```

Do you think we should add a more specific error message, e.g. instructing the user to run `make install` or use a package/container image?

> Also, we should probably add a GSG for running cilium as standalone (not in this PR).

What do you mean by standalone? Just running on a host without k8s/docker/etc.?

**Comment by brb** [comment]
@tklauser 

> Do you think we should add a more specific error message, e.g. instructing the user to run make install or use a package/container image?

Yep, something along the lines: `[..] Please run "make install-bpf"`.

> What do you mean by standalone? Just running on a host without k8s/docker/etc.?

Running cilium-agent directly on a host, i.e. not in a constainer (e.g. `./daemon/cilium-agent ...`). It doesn't matter whether k8s is provisioned or not.

**Comment by tklauser** [comment]
The following message is now logged if the BPF template directory does not exist:

```
level=fatal msg="BPF template directory: NOT OK. Please run 'make install-bpf'" error="stat /var/lib/cilium/bpf: no such file or directory" subsys=linux-datapath
```

**Comment by tklauser** [comment]
test-me-please

**Review by brb — APPROVED** [review]

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
Got hit by a change to `daemon/bpf.sha` again. Will rebase once @tgraf approves and all currently open PRs touching the file are merged.

**Review by tgraf — APPROVED** [review]

**Comment by tklauser** [comment]
test-me-please

**Comment by tgraf** [comment]
test-docs-please

**Review by aanm — CHANGES_REQUESTED** [review]
Since the Dockerfile.builder was changed this require more changes. I've created a new image in quay based on this branch. Please add a new commit (or amend existing ones) with the following tag
`quay.io/cilium/cilium-builder:2020-02-19`

Please update this tag in `Dockerfile` and `test/docker-compose.yml`

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-docs-please

**Review by aanm — APPROVED** [review]

**Pull request merged**